core/trigonometry.cpp

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#include <cmath>
+
+#include "scipp/core/except.h"
+#include "scipp/core/transform.h"
+#include "scipp/core/variable.h"
+
+namespace scipp::core {
+
+struct fail_if_variance {
+  template <class T>
+  scipp::core::detail::ValueAndVariance<T>
+  operator()(const scipp::core::detail::ValueAndVariance<T> &) const {
+    // TODO A better way for disabling operations with variances is needed.
+    throw std::runtime_error(
+        "Inverse trigonometric operation requires dimensionless input.");
+  }
+};
+
+constexpr auto inverse_trigonometric =
+    overloaded{fail_if_variance{}, [](const units::Unit &u) {
+                 expect::equals(u, units::Unit(units::dimensionless));
+                 return units::rad;
+               }};
+
+Variable acos(const Variable &var) {
```

Comment:
Should we either
- add the other functions (`cos`, `sin`, `tan`, `asin` etc...)
- have a generic way of doing this which accepts say `std::acos` as an argument
?

Reply:
I actually figured want to look into implementing trigonometrics for `Unit` first.

The other functions have been left as an exercise for the read ;)
... joking aside:
- Yes we could probably move the call to `transform` into a helper template to make future implementations easier.
- I did not plan to implement a full set of trigonometrics for 0.1, but since it is simple, maybe we could.